File: app/src/main/java/jp/juggler/fadownloader/PageRecord.java
package jp.juggler.fadownloader;

import android.app.Activity;
import android.view.View;
import android.widget.ListView;

public class PageRecord extends PagerAdapterBase.PageViewHolder{

	ListView listView;
	DownloadRecordViewer viewer;


	public PageRecord( Activity activity, View ignored ){
		super( activity, ignored );
	}

	@Override protected void onPageCreate( int page_idx, View root ) throws Throwable{
		listView = (ListView) root.findViewById( R.id.lvRecord );
		viewer = new DownloadRecordViewer();

		if( ( (ActMain) activity ).is_start ){
			onStart();
		}
	}

	@Override protected void onPageDestroy( int page_idx, View root ) throws Throwable{
		onStop();
	}

	void onStart(){
		viewer.onStart( (ActMain) activity, listView,ActMain.LOADER_ID_RECORD );
	}

	void onStop(){
		viewer.onStop();
	}
}

File: app/src/main/java/jp/juggler/fadownloader/PageRecord.kt
package jp.juggler.fadownloader

import android.app.Activity
import android.view.View
import android.widget.ListView
import jp.juggler.fadownloader.util.PagerAdapterBase

class PageRecord(activity : Activity, ignored : View) :
	PagerAdapterBase.PageViewHolder(activity, ignored) {
	
	private lateinit var listView : ListView
	internal lateinit var viewer : DownloadRecordViewer
	
	@Throws(Throwable::class)
	override fun onPageCreate(page_idx : Int, root : View) {
		listView = root.findViewById<View>(R.id.lvRecord) as ListView
		viewer = DownloadRecordViewer()
		
		if((activity as ActMain).is_start) {
			onStart()
		}
	}
	
	@Throws(Throwable::class)
	override fun onPageDestroy(page_idx : Int, root : View) {
		onStop()
	}
	
	internal fun onStart() {
		viewer.onStart(activity as ActMain, listView, ActMain.LOADER_ID_RECORD)
	}
	
	private fun onStop() {
		viewer.onStop()
	}
}
